astropy/io/misc/hdf5.py

```diff
@@ -214,6 +216,37 @@ def _encode_mixins(tbl):
     return encode_tbl
 
 
+def convert_dtype(table, character_as_bytes, copy=False):
+    """
+    Parameters:
+    -----------
+    table: `~astropy.table.Table`
+         Data table in which changes are to be made.
+    character_as_bytes: boolean
+         True if columns are to be converted to bytestrings from unicode.
+         False if columns are to be converted to unicode from bytestrings.
+    copy: boolean
+        True if changes are to be made in a new table which is copied from original table.
+    """
+    if character_as_bytes:
+        colnames = [name for name in table.colnames if table.columns[name].info.dtype.kind == 'U']
+        if colnames and copy:
+            table = table.copy(copy_data=False)
+        try:
+            table.convert_unicode_to_bytestring()
+        except (UnicodeEncodeError, AttributeError):
+            for column_name in colnames:
+                table.replace_column(column_name, np.char.encode(table.columns[column_name], "utf-8"))
+    else:
```

Comment:
`copy` parameter being ignored in this block.

Reply:
It is not required to make a copy of table in this block . Copy of Tables is made when the Table columns are in form of unicode  and the Table is to be stored in Hdf5 file .